engine/crates/wasi-component-loader/gateway-hooks.wit

```diff
@@ -72,21 +86,30 @@ interface gateway-request {
 }
 
 interface authorization {
-    use types.{error-response, shared-context};
+    use types.{error-response, shared-context, edge-definition, node-definition};
 
-    // The hook is called in the request cycle if the schema defines an authorization directive.
-    // The context is read only, and provides data written in the gateway request hook.
-    //
-    // The input is combined from the directive arguments and fields, taking the data from the 
-    // provided arguments and the field values as defined in the directive.
+    // The hook is called in the request cycle if the schema defines an authorization directive on
+    // an edge, providing the arguments of the edge and the metadata of the directive to the hook.
```

Comment:
Should we clarify that the arguments are not all arguments but only the ones selected in the directive?